execution-engine/src/wasi/strace.rs

```diff
@@ -69,23 +157,69 @@ impl Strace {
         count: u32,
         address: u32,
     ) {
-        // NOT YET IMPLEMENTED
+        if self.arg() {
+            return;
+        }
+        if !res.is_ok() {
+            eprint!("_");
+            return;
+        }
+        if let Ok(len) = memory_ref.read_u32(address) {
+            // This inefficiently copies everything, but it's only used for tracing.
+            if let Ok(bufs) = memory_ref.unpack_iovec(base, count) {
+                let mut buf: Vec<u8> = Vec::new();
```

Comment:
I think this can be rewritten as `bufs.iter().flatten().collect()`.

Reply:
I couldn't get that to work.

Reply:
What's the error?

Reply:
```
193 |                 let x = bufs.iter();
    |                              ^^^^ method not found in `IoVecSlices<'_, common::Bound<'_, <T as MemoryHandler>::Slice>>`
```

Reply:
I think you need an `as_ref()` on `bufs`.

Reply:
```
193 |                 let x = bufs.as_ref().iter().flatten();
    |                                              ^^^^^^^ `&&[u8]` is not an iterator
```

Reply:
Hmm, OK, I think we're beyond the point of marginal return trying to get this to work, lol.